sdk/storage/azure-storage-blob/src/test/java/com/azure/storage/blob/BlobTestBase.java

```diff
@@ -1073,4 +1078,159 @@ protected static TokenCredential getTokenCredential(TestMode testMode) {
+
+                try {
+                    responses.add(buildXmlPage(maxResourcesPerPage, totalPagesExpected, numberOfElementsOnThisPage,
+                        pageType, isHierarchical));
+                } catch (XMLStreamException e) {
+                    throw new RuntimeException("Failed to generate XML for paged response", e);
+                }
+            }
+            return this;
+        }
+
+        private String buildXmlPage(int maxResourcesPerPage, int totalPagesExpected, int numberOfElementsOnThisPage,
+            PageType pageType, boolean isHierarchicalForBlobs) throws XMLStreamException {
+            ByteArrayOutputStream output = new ByteArrayOutputStream();
+            XmlWriter xmlWriter = XmlWriter.toStream(output);
+
+            String elementType;
+            boolean includeDummyTagsInListElement;
+
+            switch (pageType) {
+                case LIST_BLOBS:
+                    elementType = "Blob";
+                    includeDummyTagsInListElement = false;
+                    startXml(xmlWriter, true);
+                    xmlWriter.writeStringElement("MaxResults", String.valueOf(maxResourcesPerPage));
+                    if (isHierarchicalForBlobs) {
+                        xmlWriter.writeStringElement("Delimiter", "/");
+                    }
+                    break;
+
+                case FIND_BLOBS:
+                    elementType = "Blob";
+                    includeDummyTagsInListElement = true;
+                    startXml(xmlWriter, false);
+                    xmlWriter.writeStringElement("Where", "\"dummyKey\"='dummyValue'");
+                    xmlWriter.writeStringElement("MaxResults", String.valueOf(maxResourcesPerPage));
+                    break;
+
+                case LIST_CONTAINERS:
+                    elementType = "Container";
+                    includeDummyTagsInListElement = false;
+                    startXml(xmlWriter, false);
+                    xmlWriter.writeStringElement("MaxResults", String.valueOf(maxResourcesPerPage));
+                    break;
+
+                default:
+                    throw new IllegalArgumentException("Unknown PageType: " + pageType);
+            }
+
+            writeGenericListElement(xmlWriter, elementType, numberOfElementsOnThisPage, includeDummyTagsInListElement);
+            endXml(xmlWriter, totalPagesExpected); // This calls flush
+
+            return output.toString();
+        }
+
+        private void startXml(XmlWriter xmlWriter, Boolean hasContainerAttribute) throws XMLStreamException {
+            xmlWriter.writeStartDocument();
+            xmlWriter.writeStartElement("EnumerationResults");
+            xmlWriter.writeStringAttribute("ServiceEndpoint", "https://account.blob.core.windows.net/");
+            if (hasContainerAttribute) {
+                xmlWriter.writeStringAttribute("ContainerName", "foo");
+            }
```

Comment:
Similar feedback here. It would be interesting if we could generalize this logic by exposing a general parameter such as `additionalAttributes` so we can keep the `startXml` method fairly general.